### PR TITLE
[FIX] Do not mark plugins as unavailable on conda when not running from a constructor bundle

### LINKED ISSUE #7: Plugin inproperly marked as not available to install
When install napari from https://github.com/napari/napari/pull/5743 then when I play with it I found that some of the plugins are marked as not available to install with a disabled install button. But this is my dev environment, not bundle.

image

Also, It marks the plugin that is present on conda: 

image

This error is also present in main repository, but I think that it should be fixed here.

### PR DESCRIPTION
**Pull request by jaimergp**
Closes #7 

cc @Czaki

### CODE REVIEW

File: napari_plugin_manager/_tests/test_qt_plugin_dialog.py [hunk 1]
```diff
@@ -212,6 +213,10 @@ def test_filter_not_available_plugins(plugin_dialog):
     The second plugin ("test-name-1") is available on conda-forge and
     should be enabled without the tooltip warning.
     """
+    if "no-constructor" not in request.node.name:
+        # the plugin_dialog fixture has this id
+        # skip for 'constructor' variant
+        pytest.skip()
```
Comment (Czaki) [inline napari_plugin_manager/_tests/test_qt_plugin_dialog.py:219]: This information should be put in `reason` argument of `pytest.skip`

**Review by Czaki — APPROVED** [review]
Looks ok. I prefer to have skip information passed to the skip function.

**Comment by Czaki** [comment]
Would you like to install pre-commit.ci in this repository?

**Comment by jaimergp** [comment]
> Would you like to install pre-commit.ci in this repository?

That would be nice, yes!

Also, how should I proceed to backport this to 0.4.18?

**Comment by Czaki** [comment]
> Also, how should I proceed to backport this to 0.4.18?

At this moment I do not select refactor to be backported. I need to stabilize 0.4.18 and then check if it is possible to backport. But as I understand old plugin manager (0.4.17) do not have a backend selection.

**Comment by Czaki** [comment]
@jaimergp - I have enable pre-commit on this repository.

**Comment by jaimergp** [comment]
Thanks!

**Pull request merged**